src/context.rs

```diff
@@ -1,26 +1,226 @@
-use crate::Actor;
-use std::marker::PhantomData;
+use crate::envelope::{AsyncNonReturningEnvelope, Envelope, SyncNonReturningEnvelope};
+use crate::{Actor, Address, AsyncHandler, Handler, Message};
+use crate::manager::ManagerMessage;
+#[cfg(any(doc, feature = "with-tokio-0_2", feature = "with-async_std-1"))]
+use {std::time::Duration, crate::AddressExt};
 
 /// `Context` is used to signal things to the [`ActorManager`](struct.ActorManager.html)'s
 /// management loop. Currently, it can be used to stop the actor ([`Context::stop`](struct.Context.html#method.stop)).
 pub struct Context<A: Actor> {
+    /// Whether the actor is running. It is changed by the `stop` method as a flag to the `ActorManager`
+    /// for it to call the `stopping` method on the actor
     pub(crate) running: bool,
-    phantom: PhantomData<A>, // TODO(weak_address)
+    /// The address kept by the context to allow for the `Context::address` method to work.
+    address: Address<A>,
+    /// Notifications that must be stored for immediate processing.
+    pub(crate) immediate_notifications: Vec<Box<dyn Envelope<Actor = A>>>,
```

Comment:
I hate to say it but would a `LinkedList` be a good option here? xP

Reply:
I doubt it because of the frequency of allocation required (most cases will be 2 allocs per push instead of 1) as well as slower iteration time. I think LinkedList in std lib still has to have sized elements.